src/profile-logic/import/instruments/instruments.js

```diff
@@ -0,0 +1,746 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+//types
+import type { Profile } from '../../../types/profile';
+
+//utils
+import {
+  getEmptyProfile,
+  getEmptyThread,
+} from '../../../profile-logic/data-structures';
+import BinaryPlistParser, { UID } from './BinaryPlistParser';
+import BinReader from './BinReader';
+
+// TODO make helpers.js and move the appropriate helper functions into it
+// TODO add the missing return types in the functions
+// TODO I have used console.log() statements for debug purpose, I will remove them later
+// TODO add Root frame at the top
+// TODO fill remaining meta data inside profile
+
+let fileReader;
+
+function parseBinaryPlist(bytes) {
+  const text = 'bplist00';
+  for (let i = 0; i < 8; i++) {
+    if (bytes[i] !== text.charCodeAt(i)) {
+      throw new Error('File is not a binary plist');
+    }
+  }
+
+  return new BinaryPlistParser(
+    new DataView(bytes.buffer, bytes.byteOffset, bytes.byteLength)
+  ).parseRoot();
+}
+
+function decodeUTF8(bytes: Uint8Array): string {
```

Comment:
Do you know why this function uses `escape` and `decodeURIComponent`? Are those a workaround for browsers without `TextDecoder`?

Based on the function name, I'd write this function as follows:

```javascript
const textDecoder = new TextDecoder('utf-8');
function decodeUTF8(bytes: Uint8Array): string {
  if (bytes[bytes.length - 1] === 0) {
    // Remove a single trailing null byte if present.
    return textDecoder.decode(bytes.subarray(0, -1));
  }
  return textDecoder.decode(bytes);
}
```